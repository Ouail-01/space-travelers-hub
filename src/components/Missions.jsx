function Missions() {
  return (
    <div>Mission page</div>
  );
}

export default Missions;
