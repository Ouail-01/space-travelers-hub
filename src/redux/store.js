import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    // categories: categoryReducer,
  },
});

export default store;
