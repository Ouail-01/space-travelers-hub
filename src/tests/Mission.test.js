import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Missions from '../components/Missions';

const mission = renderer.create(
  <Provider store={store}>
    <Missions />
  </Provider>,
);

describe('Mission component test', () => {
  test('Check if the missions section are available', () => {
    expect(mission).toMatchSnapshot();
  });
  test('Check if the "Description" there', () => {
    expect(mission.root.findAllByProps({ children: 'Description' })).toHaveLength(1);
  });
});

const mockMissions = [
  {
    id: '1', name: 'Mission 1', description: 'Mission 1 description', reserved: false,
  },
  {
    id: '2', name: 'Mission 2', description: 'Mission 2 description', reserved: true,
  },
];

describe('Mission component test', () => {
  test('renders mission id correctly', () => {
    expect(mockMissions[0].id).toBe('1');
  });
  test('renders mission name correctly', () => {
    expect(mockMissions[1].name).toBe('Mission 2');
  });
  test('renders mission reservation correctly', () => {
    expect(mockMissions[0].reserved).toBe(false);
  });
});
