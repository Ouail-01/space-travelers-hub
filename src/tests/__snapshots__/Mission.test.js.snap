// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component test Check if the missions section are available 1`] = `
<div
  className="missionContainer"
>
  <table
    className="missionTable"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
